### ISSUE TITLE
Submenus for ContextMenu's aren't documented

### ISSUE THREAD

**Issue by deto** [handsontable/docs-pro#4]
As per title, could really use some documentation on using submenus when creating a custom context menu.

The documentation states to use the Jquery contextMenu documentation for details, but the example here:

https://swisnl.github.io/jQuery-contextMenu/demo/sub-menus.html

Doesn't work.

Looking at this Issue, it appears special support for submenu's was added here:

https://github.com/handsontable/handsontable/issues/1350

But there is no documentation detailing the correct syntax.  Only thing I could find was this stackoverflow answer here:

http://stackoverflow.com/questions/24074313/how-create-a-submenu-in-a-contextmenu-handsontable

Which still took me a while to get working because I didn't realize the sub-menu item keys are required to be prefixed with the parent menu key.

**Comment by AMBudnik** [comment]
Thanks for sharing @deto 
Our submenu tutorial can be found at: https://docs.handsontable.com/pro/1.4.1/demo-context-menu.html (context menu section) but you're right - we can add more information and/or examples.

ps. please notice that there are two bugs related that you can came accross by creating your own submenu (https://github.com/handsontable/handsontable/issues/3550 and https://github.com/handsontable/handsontable/issues/3533)

**Comment by deto** [comment]
Maybe I'm missing it, but on the page you linked, I can't find a single mention of sub-menus.  To clarify, by sub-menus, I don't mean context menus, but rather the ability to do a hierarchical context menu - where you hover over one option and another set of choices pops out.  This is shown in the "alignment" example on the page you linked, but since that is a built-in option, there is no instruction on how to configure your own sub-menu.

It appears this is an option, though, based on the issue I posted above where the functionality was added.  I just had to go digging around SO to find the right syntax, and even then, there were other caveats to get it working - none that are a big deal, but were just time-consuming to figure out without documentation.

**Comment by AMBudnik** [comment]
Wow.. I was sure that it's already there! Sorry for a misguidance.
We surely have to add this to the docs... even if it's not working as expected (like in the https://github.com/handsontable/handsontable/issues/3550)

**Comment by deto** [comment]
Yep - figured it was just an oversight and that you'd want to have some documentation on it (since someone spent time coding up the feature!)
